drivers/storage/portworx/components_test.go

```diff
@@ -2662,6 +2662,76 @@ func TestAutopilotInvalidCPU(t *testing.T) {
 		fmt.Sprintf("%v %v Failed to setup Autopilot.", v1.EventTypeWarning, util.FailedComponentReason))
 }
 
+func TestSecurityInstall(t *testing.T) {
+	coreops.SetInstance(coreops.New(fakek8sclient.NewSimpleClientset()))
+	reregisterComponents()
+	k8sClient := testutil.FakeK8sClient()
+	driver := portworx{}
+	driver.Init(k8sClient, runtime.NewScheme(), record.NewFakeRecorder(0))
+
+	cluster := &corev1alpha1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-cluster",
+			Namespace: "kube-test",
+		},
+		Spec: corev1alpha1.StorageClusterSpec{
+			Security: &corev1alpha1.SecuritySpec{
+				Enabled: true,
+			},
+		},
+	}
+
+	err := driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	// Default configuration should be loaded with autogenerated secret keys. ConfigMap
+	securitySecret := &v1.Secret{}
+	err = testutil.Get(k8sClient, securitySecret, pxutil.SecurityPXAuthKeysSecretName, cluster.Namespace)
+	require.NoError(t, err)
+
+	require.Equal(t, 64, len(securitySecret.StringData[component.SecuritySystemSecretKey]))
+	require.Equal(t, 64, len(securitySecret.StringData[component.SecuritySharedSecretKey]))
+	require.NotEqual(t, securitySecret.StringData[component.SecuritySystemSecretKey], securitySecret.StringData[component.SecuritySharedSecretKey])
+
+	// User pre-configured values should not be overwritten
+	err = testutil.Delete(k8sClient, securitySecret)
+	require.NoError(t, err)
+	systemSecretKey := "systemSecretKey"
+	sharedSecretKey := "sharedSecretKey"
+	cluster.Spec.Env = append(cluster.Spec.Env, v1.EnvVar{
+		Name:  pxutil.EnvKeyPortworxAuthSystemKey,
+		Value: systemSecretKey,
+	})
+	cluster.Spec.Env = append(cluster.Spec.Env, v1.EnvVar{
+		Name:  pxutil.EnvKeyPortworxAuthJwtSharedSecret,
+		Value: sharedSecretKey,
+	})
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	securitySecret = &v1.Secret{}
+	err = testutil.Get(k8sClient, securitySecret, pxutil.SecurityPXAuthKeysSecretName, cluster.Namespace)
+	require.NoError(t, err)
+	require.Equal(t, systemSecretKey, securitySecret.StringData[component.SecuritySystemSecretKey])
+	require.Equal(t, sharedSecretKey, securitySecret.StringData[component.SecuritySharedSecretKey])
+
+	// Disable security, secret should not be deleted.
+	cluster.Spec.Security.Enabled = false
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+	err = testutil.Get(k8sClient, securitySecret, pxutil.SecurityPXAuthKeysSecretName, cluster.Namespace)
+	require.NoError(t, err)
+
+	// Delete cluster (cluster wipe), secret should be deleted.
```

Comment:
In this case secret should not be deleted (the assertion is correct, just the comment is incorrect)
Let's add a case where the deleteStrategy is set and ensure that the secret is deleted.